src/main/java/com/cloudbees/jenkins/plugins/bitbucket/hooks/NativeServerPushHookProcessor.java

```diff
@@ -44,6 +44,9 @@
 @RestrictedSince("933.3.0")
 public class NativeServerPushHookProcessor extends HookProcessor {
 
+    private static final boolean SCAN_ON_PUSH_WITH_EMPTY_CHANGES = Boolean.getBoolean(
+        NativeServerPushHookProcessor.class.getName() + ".scanOnPushWithEmptyChanges");
```

Comment:
NIT: Use a [SystemProperty](https://javadoc.jenkins.io/jenkins/util/SystemProperties.html)?  (not everyone can set System properties).
Should this default to false (ie invert the setting so you disable full scans for empty changes rather than the other way around?)

Reply:
Done.